newsroom/users/service.py

```diff
@@ -0,0 +1,181 @@
+from typing import Any, Dict
+
+from quart_babel import gettext
+from werkzeug.exceptions import BadRequest
+
+from superdesk.core import get_app_config
+from superdesk.flask import request, abort, session
+from superdesk.utils import is_hashed, get_hash
+from superdesk.core.resources.service import ResourceModelType
+
+from newsroom.settings import get_setting
+from newsroom.auth.utils import (
+    get_company_auth_provider,
+    is_current_user_admin,
+    is_current_user_account_mgr,
+    is_current_user_company_admin,
+    send_token,
+)
+from newsroom.core import get_current_wsgi_app
+from newsroom.signals import user_created, user_updated, user_deleted
+from newsroom.core.resources.service import NewshubAsyncResourceService
+from newsroom.companies.utils import get_updated_products, get_updated_sections
+
+from .model import UserResourceModel
+from .users import COMPANY_ADMIN_ALLOWED_PRODUCT_UPDATES, COMPANY_ADMIN_ALLOWED_UPDATES, USER_PROFILE_UPDATES
+
+
+class UsersService(NewshubAsyncResourceService[UserResourceModel]):
+    resource_name = "users"
+
+    def get_model_instance_from_dict(self, data: Dict[str, Any]) -> ResourceModelType:
```

Comment:
Rather than supported invalid data, what if we have a data update script that fixes the data. That way we don't need this custom code to support invalid data in the DB

Reply:
Yeah, that makes sense. Do you mean like those from this command `python manage.py data:generate_update`? If so, yes we should do it. Although I'm not sure if these commands are working at the moment. Perhaps we should have this in a separate small ticket (also to remove this workaround). Thoughts?

Reply:
Yes that's the one. Those commands should work, and seem to work on my local machine (although we're missing `manage.py` in newsroom-core repo, so I generally run it on a `newsroom-<name>-app` repo and copy the file over.

If you can, have a quick go at it, otherwise I will create a ticket to cover this (and test the commands).

Reply:
Fixed in commits: https://github.com/superdesk/newsroom-core/pull/1036/commits/d437a72dc5ab0378ecb89db8e61a853c2b947dbb https://github.com/superdesk/newsroom-core/pull/1036/commits/c54ddd987aba4988563035cb8ba016218087a64b

You would need to run this below to make it work properly. 
```python manage.py data_upgrade -i 00016_20240820-184145_users```